File: src/main/java/no/nav/domain/pensjon/regler/beregning2011/EgenopptjentUforetrygd.java
package no.nav.domain.pensjon.regler.beregning2011;

import no.nav.domain.pensjon.regler.kode.FormelKodeCti;
import no.nav.domain.pensjon.regler.trygdetid.AnvendtTrygdetid;

import java.io.Serializable;

/**
 * @author Steinar Hjellvik (Decisive) - PKPYTON-1746
 *
 */
public class EgenopptjentUforetrygd implements Serializable {

    

    private double faktor;

    private FormelKodeCti formelKode;

    private int arsbelop;

    private AbstraktBeregningsgrunnlag beregningsgrunnlagOrdiner;

    private AbstraktBeregningsgrunnlag beregningsgrunnlagYrkesskade;

    private boolean beregningsgrunnlagYrkesskadeBest;

    /**
     * Prosentsats brukt for påslag.
     */
    private int konverteringsPaslagForRedGP;

    /**
     * Påslag pga økt redusert grunnpensjon sats til egenopptjent uføretrygd som angår konvertert uføretidpunkt.
     */
    private int konverteringsPaslagForRedGPSats;


    /**
     * Trygdetid som er brukt ved beregning av egenopptjent uføretrygd.
     */
    private AnvendtTrygdetid anvendtTrygdetid;

}

File: src/main/kotlin/no/nav/pensjon/regler/domain/beregning2011/EgenopptjentUforetrygd.kt
package no.nav.pensjon.regler.domain.beregning2011

import no.nav.pensjon.regler.domain.enum.FormelKodeEnum
import no.nav.pensjon.regler.domain.trygdetid.AnvendtTrygdetid
import java.io.Serializable

class EgenopptjentUforetrygd : Serializable {
    var faktor = 0.66
    var formelKodeEnum: FormelKodeEnum? = null
    var arsbelop = 0
    var beregningsgrunnlagOrdiner: AbstraktBeregningsgrunnlag? = null
        set(aBeregningsgrunnlag) {
            if (aBeregningsgrunnlag is BeregningsgrunnlagOrdiner
                || aBeregningsgrunnlag is BeregningsgrunnlagKonvertert
            ) {
                field = aBeregningsgrunnlag
            }
        }
    var beregningsgrunnlagYrkesskade: AbstraktBeregningsgrunnlag? = null
        set(aBeregningsgrunnlag) {
            if (aBeregningsgrunnlag is BeregningsgrunnlagYrkesskade
                || aBeregningsgrunnlag is BeregningsgrunnlagKonvertert
            ) {
                field = aBeregningsgrunnlag
            }
        }
    var beregningsgrunnlagYrkesskadeBest = false

    /**
     * Prosentsats brukt for Påslag.
     */
    var konverteringsPaslagForRedGP = 0

    /**
     * Påslag pga økt redusert grunnpensjon sats til egenopptjent uføretrygd som angår konvertert Uføretidpunkt.
     */
    var konverteringsPaslagForRedGPSats = 0

    /**
     * Trygdetid som er brukt ved beregning av egenopptjent uføretrygd.
     */
    var anvendtTrygdetid: AnvendtTrygdetid? = null
}